### ISSUE TITLE
Incorrect label vertical alignment for certain font sizes

### ISSUE THREAD

**Issue by marisn** [OSGeo/grass#2854]
A row of vector points has labels shown next to them. Depending on the font size, some labels will be vertically shifted or all labels will be on the same line (as expected).
See screenshot with only one difference – for one row font size is set to 21px and for other – 22px. Vertical alignment should be identical for all letters but is different for the 21px case.
![21px](https://user-images.githubusercontent.com/17123023/220985347-ef6829ea-36bd-4099-9c28-0eaae8a9f7ee.png)

**Comment by nilason** [comment]
I have tried to reproduce this with no success. Lines up nicely with various sizes, font types...

**Comment by marisn** [comment]
On my system this happens with all fonts only the size needs to be adjusted as for each font sizes that trigger the issue differ. I tried both cairo and PNG renderers and both are affected. This issue has been reproduced on another Linux system too. My Debian uses libfreetype6-2.12.1
This is how letters with bottom left alignment look like:
![image](https://user-images.githubusercontent.com/17123023/221134669-5c5542be-bb93-4873-afad-6cc1fb4cb13e.png)

**Comment by nilason** [comment]
This is what I've been using (and variants of it):

```sh
#!/bin/sh

echo "1|629880|222510|a
2|629880|222512|b
3|629880|222514|c
4|629880|222516|d
5|629880|222518|e
6|629880|222520|f
7|629880|222522|g
8|629880|222524|h
9|629880|222526|i
10|629880|222528|1
11|629880|222530|2
12|629880|222532|3
13|629880|222534|4
14|629880|222536|5
15|629880|222538|6
16|629880|222540|7
17|629880|222542|8
18|629880|222544|9" | v.in.ascii input=- output=labelmap --overwrite

g.region vector=labelmap -p

d.mon start=wx0

d.vect map=labelmap color=0:29:57:255 width=1 \
    attribute_column=str_1 label_bcolor=13:0:200:255 \
    label_size=21 \
    yref=center xref=left
```

I now realise the `yref` and `xref` seem not to take effect for me.

**Comment by marisn** [comment]
> I now realise the `yref` and `xref` seem not to take effect for me.

That will depend on size of computational region, font size, output size and phase of the moon. With your data I managed to get even so ridiculous output:
![image](https://user-images.githubusercontent.com/17123023/221152737-af87b31a-810d-457c-a114-2949f2911fe5.png)

**Comment by nilason** [comment]
> That will depend on size of computational region, font size, output size and phase of the moon. With your data I managed to get even so ridiculous output:

Not to forget which side you woke up on that particular morning :-) 

I've been testing in nc_spm_full_v2alpha2, this is what it looks like for me:

![out](https://user-images.githubusercontent.com/14186207/221156891-9b8fc2a9-7fdd-482f-ae13-880f980cf26e.png)

**Comment by nilason** [comment]
Interestingly, there was a ticket related to my particular issue 11 years ago: https://trac.osgeo.org/grass/ticket/1667

**Comment by marisn** [comment]
The logic of alignment in d.vect is totally bogus. If I disable Yoffset, the issue is gone https://github.com/OSGeo/grass/blob/50815f96233e31895f333e8549078b5295ae2fe1/display/d.vect/label.c#L172

**Comment by veroandreo** [comment]
> > That will depend on size of computational region, font size, output size and phase of the moon. With your data I managed to get even so ridiculous output:
> 
> Not to forget which side you woke up on that particular morning :-)
> 
> I've been testing in nc_spm_full_v2alpha2, this is what it looks like for me:
> 
> ![out](https://user-images.githubusercontent.com/14186207/221156891-9b8fc2a9-7fdd-482f-ae13-880f980cf26e.png)

I get something similar in Fedora 36. No effect of `yref` and `xref`, no matter what I put there, the output is the same.

**Comment by nilason** [comment]
I have now corrected v.in.ascii and g.region in above example script, and now alignment seems to work.

**Comment by wenzeslaus** [comment]
> I have now corrected v.in.ascii and g.region in above example script, and now alignment seems to work.

@nilason Can you please share your corrected code so we can test #2857? Script or notebook in the source code might be good basis for testing in the future.

**Comment by nilason** [comment]
> > I have now corrected v.in.ascii and g.region in above example script, and now alignment seems to work.
> 
> @nilason Can you please share your corrected code so we can test #2857? Script or notebook in the source code might be good basis for testing in the future.

I referred to the testing script above (https://github.com/OSGeo/grass/issues/2854#issuecomment-1443295725). Now I have hopefully corrected/updated it in-place. (Initially made some last minute correction without testing when I posted it).

Did some further testing, sometimes it works, sometimes not. I even hade the jumpy result as @marisn . But I haven't been able to see some pattern.
